metaflow/plugins/cards/card_cli.py

```diff
@@ -0,0 +1,388 @@
+    path_tuples = card_datastore.get_card_names(card_paths)
+    ctx.obj.echo(
+        "\nFound %d card matching for your query..." % len(path_tuples), fg="green"
+    )
+    task_pathspec = "/".join(path_spec.split("/")[1:])
+    card_list = []
+    for path_tuple in path_tuples:
+        card_name = "%s-%s" % (path_tuple.type, path_tuple.hash)
+        card_list.append(card_name)
+
+    random_idx = 0 if len(path_tuples) == 1 else random.randint(0, len(path_tuples) - 1)
+    _, randhash = path_tuples[random_idx]
+    ctx.obj.echo("\n\t".join([""] + card_list), fg="blue")
+    ctx.obj.echo(
+        "\n\tExample access from CLI via: \n\t %s\n"
+        % make_command(
+            scriptname,
+            task_pathspec,
+            command=command,
+            hash=randhash[:NUM_SHORT_HASH_CHARS],
+        ),
+        fg="yellow",
+    )
+
+
+def make_command(
+    script_name,
+    taskspec,
+    command="get",
+    hash=None,
+):
+    calling_args = ["--hash", hash]
+    return " ".join(
+        [
+            ">>>",
+            "python",
+            script_name,
+            "card",
+            command,
+            taskspec,
+        ]
+        + calling_args
+    )
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="Commands related to @card decorator.")
+@click.pass_context
+def card(ctx):
+    # We set the metadata values here so that top level arguments to --datastore and --metadata
+    # Can work with the Metaflow client.
+    # If we don't set the metadata here than the metaflow client picks the defaults when calling the `Task`/`Run` objects. These defaults can come from the `config.json` file or based on the `METAFLOW_PROFILE`
+    from metaflow import metadata
+
+    setting_metadata = "@".join(
+        [ctx.obj.metadata.TYPE, ctx.obj.metadata.default_info()]
+    )
+    metadata(setting_metadata)
+    # set the card root to the datastore according to the configuration.
+    root_pth = CardDatastore.get_storage_root(ctx.obj.flow_datastore._storage_impl.TYPE)
+    if root_pth is not None:
+        ctx.obj.flow_datastore._storage_impl.datastore_root = root_pth
+
+
+def card_read_options_and_arguments(func):
+    @click.option(
+        "--hash",
+        default=None,
+        show_default=True,
+        type=str,
+        help="Hash of the stored HTML",
+    )
+    @click.option(
+        "--type",
+        default=None,
+        show_default=True,
+        type=str,
+        help="Type of card being created",
+    )
+    @click.option(
+        "--dont-follow-resumed",
+        default=False,
+        is_flag=True,
+        help="Doesn't follow the origin-task-id of resumed tasks to seek cards stored for resumed tasks.",
+    )
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def render_card(mf_card, task, timeout_value=None):
+    rendered_info = None
+    if timeout_value is None or timeout_value < 0:
```

Comment:
do we want a default timeout of None then?

Reply:
default is about 45 seconds. This is when the user explicitly gives `timeout=None` in the decorator.

Reply:
Ah, so the timeout is somewhere else? This function makes it look like the timeout is None by default.

Reply:
Yes timeout comes from the subprocess that invokes the cli create command.